File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        count = []
        sum_ = 0
        res = 0
        for i in range(len(nums)):
            sum_ += nums[i]
        for i in range(len(nums)):
            count.append(sum_)
            sum_ -= nums[i]
        left = nums[0]
        for i in range(1, len(count)):
            if left >= count[i]:
                res += 1
            left += nums[i]
        return res

File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var waysToSplitArray = function (nums) {
    let count = []
    let sum = 0
    let res = 0
    for (let i = 0; i < nums.length; i++) {
        sum += nums[i]
    }
    for (let i = 0; i < nums.length; i++) {
        count.push(sum)
        sum -= nums[i]
    }
    let left = nums[0]
    for (let i = 1; i < count.length; i++) {
        if (left >= count[i]) {
            res += 1
        }
        left += nums[i]
    }
    return res
};